script/tool/lib/src/common/core.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:colorize/colorize.dart';
+import 'package:file/file.dart';
+import 'package:yaml/yaml.dart';
+
+/// The signature for a print handler for commands that allow overriding the
+/// print destination.
+typedef Print = void Function(Object? object);
+
+/// Key for windows platform.
+const String kPlatformFlagWindows = 'windows';
+
+/// Key for macos platform.
+const String kPlatformFlagMacos = 'macos';
+
+/// Key for linux platform.
+const String kPlatformFlagLinux = 'linux';
+
+/// Key for IPA (iOS) platform.
+const String kPlatformFlagIos = 'ios';
+
+/// Key for APK (Android) platform.
+const String kPlatformFlagAndroid = 'android';
+
+/// Key for Web platform.
+const String kPlatformFlagWeb = 'web';
+
+/// Key for enable experiment.
+const String kEnableExperiment = 'enable-experiment';
+
+/// Returns whether the given directory contains a Flutter package.
+bool isFlutterPackage(FileSystemEntity entity) {
+  if (entity is! Directory) {
+    return false;
+  }
+
+  try {
+    final File pubspecFile = entity.childFile('pubspec.yaml');
+    final YamlMap pubspecYaml =
+        loadYaml(pubspecFile.readAsStringSync()) as YamlMap;
+    final YamlMap? dependencies = pubspecYaml['dependencies'] as YamlMap?;
+    if (dependencies == null) {
+      return false;
+    }
+    return dependencies.containsKey('flutter');
+  } on FileSystemException {
+    return false;
+  } on YamlException {
+    return false;
+  }
+}
+
+/// Prints `errorMessage` in red.
+void printError(String errorMessage) {
+  final Colorize redError = Colorize(errorMessage)..red();
+  print(redError);
+}
```

Comment:
What makes these methods not in the plugin_util.dart?

Reply:
They aren't related to plugins. One of the goals of the refactoring is to stop using "plugin" to mean "package" in most places in this tool, since it's used for both flutter/plugins and flutter/packages.

Reply:
Ah I see. So I think a future refactoring that updates the class `PluginCommand` to a different name is coming?

Reply:
Yes, and flag renaming, and various other things.